sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -81,15 +85,26 @@ public HttpTransport(
     this.readTimeout = readTimeoutMillis;
     this.options = options;
     this.bypassSecurity = bypassSecurity;
-    this.sentryUrl = sentryUrl;
+
+    try {
+      final URI uri = sentryUrl.toURI();
```

Comment:
@bruno-garcia ideally I'd get this `URI` in the ctor instead of the current `URL`, but I'd not like to break the public API, I think this would be fine here.
what do you think?

Reply:
yeah this looks good for now but worth adding a task for v3.0.0 to refactor this. Perhaps we need to revisit the `Dsn` type (like add a `toStore` and `toEnvelope` or something completely different).
Such discussion for sure woudl involve the whole SDK team